### PR TITLE
ARROW-15778: [Java] set native endian to schema

### PR DESCRIPTION
**Pull request by kiszk**
This PR fixes a problem that endianness field in schema was not emitted in IPC stream.

I will create another JIRA/PR for adding test due to the current limited bandwidth.

### CODE REVIEW

**Comment by kiszk** [comment]
cc @pitrou

File: java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java [hunk 1]
```diff
@@ -207,6 +209,7 @@ public int getSchema(FlatBufferBuilder builder) {
     int fieldsOffset = org.apache.arrow.flatbuf.Schema.createFieldsVector(builder, fieldOffsets);
     int metadataOffset = FBSerializables.writeKeyValues(builder, metadata);
     org.apache.arrow.flatbuf.Schema.startSchema(builder);
+    org.apache.arrow.flatbuf.Schema.addEndianness(builder, (short) (LITTLE_ENDIAN ? 0 : 1));
```
Comment (lidavidm) [inline java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java:212]: nit: we could use the generated constants for this? [org.apache.arrow.flatbuf.Endianness](https://arrow.apache.org/docs/java/reference/org/apache/arrow/flatbuf/Endianness.html)

**Comment by BryanCutler** [comment]
@github-actions crossbow submit integration

**Comment by BryanCutler** [comment]
@github-actions crossbow submit -g integration

**Review by BryanCutler — APPROVED** [review]
LGTM. From what I could tell, the turbodbc integration test failures were due to build issues and unrelated.

**Comment by BryanCutler** [comment]
merged to master, thanks @kiszk

**Pull request closed without merging**